src/transports/sentry_transport_curl.c

```diff
@@ -218,6 +221,9 @@ sentry__curl_send_task(void *_envelope, void *_state)
     if (state->http_proxy) {
         curl_easy_setopt(curl, CURLOPT_PROXY, state->http_proxy);
     }
+    if (state->socks5_proxy) {
```

Comment:
Should we disallow having both http and socks5 proxies (e.g. by having one take priority over the other if both are given), or can these work in tandem?

Reply:
No, you are right. We should prevent double proxy configs at the options interface (i.e., reset an HTTP proxy when setting a SOCKS5 proxy and vice versa). 

The HTTP proxy option results in a `CONNECT` method proxy, a persistent tunnel, and thus a stateful connection. We can't have two persistent tunnels (that would require two clients).

Please also consider providing a single proxy option (where we keep but deprecate the current HTTP-only proxy option or reuse it with sufficient documentation). In this option, we decide, based on the URI, which proxy configuration we attempt when configuring our HTTP clients. 

This is more generic and extensible and provides the convenience of not carrying around two (or, in the future, more) mutually exclusive config items.

Reply:
@supervacuus I've added a single proxy option & kept the old API methods (which now just call to the new `sentry_options_set/get_proxy` function). I think that should resolve this comment?